sklearn/tests/test_common.py

```diff
@@ -63,8 +65,14 @@ def test_non_meta_estimators():
             continue
         if name.startswith("_"):
             continue
-        for check in _yield_all_checks(name, Estimator):
-            yield _named_check(check, name), name, Estimator
+        estimator = Estimator()
+        # check this on class
+        yield _named_check(
+            check_no_fit_attributes_set_in_init, name), name, Estimator
+
+        for check in _yield_all_checks(name, estimator):
+            set_testing_parameters(estimator)
```

Comment:
hm... it seems pretty unexpected to me, and the parameters are really arbitrary.... I guess the user can always get back control by passing instances